src/lpython/parser/semantics.h

```diff
@@ -798,8 +798,8 @@ static inline ast_t* concat_string(Allocator &al, Location &l,
         x.c_str(p.m_a), expr_contextType::Load)
 // `x.int_n` is of type BigInt but we store the int64_t directly in AST
 #define INTEGER(x, l) make_ConstantInt_t(p.m_a, l, x, nullptr)
-#define STRING1(x, l) make_ConstantStr_t(p.m_a, l, unescape_string(p.m_a, x), nullptr)
-#define STRING2(x, y, l) concat_string(p.m_a, l, EXPR(x), unescape_string(p.m_a, y), nullptr)
+#define STRING1(x, l) make_ConstantStr_t(p.m_a, l, LCompilers::s2c(p.m_a, LCompilers::unescape_string(p.m_a, x.str())), nullptr)
```

Comment:
The changes in this PR need to be updated here, I think. https://github.com/lcompilers/lpython/pull/1762/files#diff-5a134f23f49700449fdd6d749a06b72ca2c844ed6c30fa752d6194b4b8b5efb7

Reply:
Could you please explain? I think the only change here is in the API of `unescape_string`. It used to return `char*` in LPython but when synced with LFortran it returns `std::string`, so its usage has to be updated everywhere in LPython.

Reply:
LFortran uses this function in asr_to_llvm, I think that needs to be reverted.
So, the changes related to unescape can be reverted.

Reply:
See https://github.com/lcompilers/lpython/pull/1502#issuecomment-1535080906